### PR TITLE
post: cli 1.3

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+author: cexbrayat
+tags: ["Angular 2", "Angular", "Angular 4", "Angular CLI"]
+description: "Angular 4.3 is out! Which new features are included?"
+---
+
+[Angular CLI 1.3.0](https://github.com/angular/angular-cli/releases/tag/v1.3.0) is out with some nice new features!
+
+It now officially supports ES2017 and TypeScript 2.4 and is ready for Angular 5 (which should be out shortly).
+
+You can check out what files you need to update to upgrade using [angular-cli-diff](https://github.com/cexbrayat/angular-cli-diff),
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: to update to upgrade -> repetition

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+## Bundle sizes
+
+The Angular team is putting a ton of work to generate the smallest bundles possible.
+Angular CLI uses Webpack 3 and now leverages its [scope hoisting feature](https://medium.com/webpack/webpack-3-official-release-15fd2dd8f07b)
+which resulted in some nice gains, but there is still room for improvements.
+
+The Angular team is working on a better tree-shaking with a new project [`@angular/devkit`](https://github.com/angular/devkit), that the CLI now uses.
+It's a bit experimental, so it's behind a new flag called `--build-optimizer`.
+It does some manipulations on your compiled code, like folding the static properties inside the class,
+scrubbing the Angular specific decorators and metadata (that are unneeded when you have compiled in AoT mode), adds a `/*@__PURE__*/` comment on classes, etc.
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: adds -> adding

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+And all that gives us smaller bundles \o/
+I tested it on a medium application:
+
+Size without `--build-optimizer`: 871K (211K gzip)
+
+With `--build-optimizer`: 778K (192K gzip)
+
+That's a small but nice win.
+But for others it resulted in massive gains!
+So it can be interested for you, especially if you are using Angular  Material
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: Angular  Material -> Angular Material (1 space)

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+(check the comments of [this issue](https://github.com/angular/material2/issues/4137)).
+
+You can read more about how this feature works [here](https://github.com/angular/devkit/tree/master/packages/angular_devkit/build_optimizer).
+Be warn that it is experimental though, and can break your app, so be careful...
+
+## Named chunks
+
+While we are talking about building the application,
+a new flag, called `--named-chunks`, will generate nicely named chunks.
+Until now, if you had lazy-loaded module in your application,
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: module -> modules

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+the bundles looked like that:
+
+    main.bundle.js
+    0.chunk.js
+    1.chunk.js
+
+With this new flag, the CLI attempts to name it with the name of the Angular module:
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: module -> modules

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    main.bundle.js
+    users.module.chunk.js
+    admin.module.chunk.js
+
+This is the default in development so you don't even have to add the flag to your command,
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: development -> development mode

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+but it's not the default in production mode.
+
+## Server side rendering
+
+The SSR is getting better and better,
+and the CLI now has a documentation to guide you to set it up,
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: to guide you to set it up ->  to guide you in setting it up

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+check it out: [universal rendering](https://github.com/angular/angular-cli/wiki/stories-universal-rendering).
+It's honestly still very early days, and won't work with most apps (no lazy-loading support for example).
+
+## Proxy configuration
+
+We are able to define a proxy configuration since some time now,
```
Comment (acrepet) [inline _posts/2017-08-10-angular-cli-1.3.md]: since some time now -> since some times now

Comment (cexbrayat) [inline _posts/2017-08-10-angular-cli-1.3.md]: I think it's some time here, but I may be wrong.
We'll see what @jnizet says

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: What's new in Angular CLI 1.3?
+author: cexbrayat
+tags: ["Angular 2", "Angular", "Angular 4", "Angular CLI"]
+description: "Angular 4.3 is out! Which new features are included?"
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: Wrong title

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+The Angular team is working on a better tree-shaking with a new project [`@angular/devkit`](https://github.com/angular/devkit), that the CLI now uses.
+It's a bit experimental, so it's behind a new flag called `--build-optimizer`.
+It does some manipulations on your compiled code, like folding the static properties inside the class,
+scrubbing the Angular specific decorators and metadata (that are unneeded when you have compiled in AoT mode), adding a `/*@__PURE__*/` comment on classes, etc.
+All these actions aim for the same goal: helping Webpack and Uglify to do a better tree-shaking
+and a better dead code elimination.
+
+And all that gives us smaller bundles \o/
+I tested it on a medium application:
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: medium-size?

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+With `--build-optimizer`: 778K (192K gzip)
+
+That's a small but nice win.
+But for others it resulted in massive gains!
+So it can be interested for you, especially if you are using Angular Material
+(check the comments of [this issue](https://github.com/angular/material2/issues/4137)).
+
+You can read more about how this feature works [here](https://github.com/angular/devkit/tree/master/packages/angular_devkit/build_optimizer).
+Be warn that it is experimental though, and can break your app, so be careful...
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: warned

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+While we are talking about building the application,
+a new flag, called `--named-chunks`, will generate nicely named chunks.
+Until now, if you had lazy-loaded modules in your application,
+the bundles looked like that:
+
+    main.bundle.js
+    0.chunk.js
+    1.chunk.js
+
+With this new flag, the CLI attempts to name it with the names of the Angular modules:
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: Why "attempts"? names it

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    users.module.chunk.js
+    admin.module.chunk.js
+
+This is the default in development mode so you don't even have to add the flag to your command,
+but it's not the default in production mode.
+
+## Server side rendering
+
+The SSR is getting better and better,
+and the CLI now has a documentation to guide you,
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: a piece of documentation

File: _posts/2017-08-10-angular-cli-1.3.md (hunk 8 above)
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: Agnès doit être bruxelloise: some times -> quelques fois. some time: quelque temps. I agree with Cedric.

Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: I'd say: It's possible to define

File: _posts/2017-08-10-angular-cli-1.3.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+check it out: [universal rendering](https://github.com/angular/angular-cli/wiki/stories-universal-rendering).
+It's honestly still very early days, and won't work with most apps (no lazy-loading support for example).
+
+## Proxy configuration
+
+We are able to define a proxy configuration since some time now,
+and it's quite useful as you can for example redirect all the calls to an API
```
Comment (jnizet) [inline _posts/2017-08-10-angular-cli-1.3.md]: redirect all the calls -> forward all the requests

**Pull request merged**